BinSkim Error BA2007 in react-native-win32.dll

### Problem Description

Seeing this in publish - we need to stop ignoring warning 4244.

```
##[error]1. BinSkim Error BA2007 - File: vnext/target/x64/Debug/React.Windows.Desktop.DLL/react-native-win32.dll.  
Signature: f361a328ed2845ecea35845e2aa765efd85c5cfffe99e4658d6eb504eefc7aa1
Tool: BinSkim: Rule: BA2007 (EnableCriticalCompilerWarnings). https://github.com/microsoft/binskim/blob/main/docs/BinSkimRules.md#rule-BA2007EnableCriticalCompilerWarnings
'react-native-win32.dll' disables compiler warning(s) which are required by policy. A compiler warning is typically required if it has a high likelihood of flagging memory corruption, information disclosure, or double-free vulnerabilities. To resolve this issue, enable the indicated warning(s) by removing /Wxxxx switches (where xxxx is a warning id indicated here) from your command line, and resolve any warnings subsequently raised during compilation. An example compiler command line triggering this check was: -c -ID:\a\_work\_temp\.nuget\packages\microsoft.windowsappsdk\1.6.240923002\include -ID:\a\_work\_temp\.nuget\packages\microsoft.windowsappsdk\1.6.240923002\include\ -ID:\a\_work\_temp\.nuget\packages\microsoft.web.webview2\1.0.2651.64\build\native\include-winrt\ -ID:\a\_work\_temp\.nuget\packages\microsoft.web.webview2\1.0.2651.64\build\native\include\ -ID:\a\_work\_temp\.nuget\packages\boost\1.83.0\lib\native\include\ -ID:\a\_work\_temp\.nuget\packages\reactnative.v8jsi.windows\0.71.8\build\native\include -ID:\a\_work\1\s\vnext\Common -ID:\a\_work\1\s\vnext\stubs -ID:\a\_work\1\s\vnext\include -ID:\a\_work\1\s\vnext\Shared -ID:\a\_work\1\s\vnext\include\Shared -ID:\a\_work\1\s\node_modules\react-native\ReactCommon\yoga -ID:\a\_work\1\s\vnext\Microsoft.ReactNative -ID:\a\_work\1\s\vnext\Microsoft.ReactNative\Fabric\platform -ID:\a\_work\1\s\vnext\Microsoft.ReactNative\ReactHost -ID:\a\_work\1\s\vnext\Microsoft.ReactNative\Views -ID:\a\_work\1\s\vnext\codegen -ID:\a\_work\1\s\vnext\codegen\react\components\rnwcore -ID:\a\_work\1\s\node_modules\react-native\ReactCommon\react\renderer\graphics\platform\cxx -ID:\a\_work\1\s\vnext\Microsoft.ReactNative.Cxx\ -ID:\a\_work\1\s\vnext\Microsoft.ReactNative.Cxx\ -ID:\a\_work\1\s\vnext\Microsoft.ReactNative.Cxx\ -ID:\a\_work\1\s\vnext\Microsoft.ReactNative.Cxx\ -ID:\a\_work\1\s\vnext\Microsoft.ReactNative.Cxx\ReactCommon -ID:\a\_work\1\s\vnext\Microsoft.ReactNative.Cxx\stubs -ID:\a\_work\1\s\node_modules\.node-api-jsi\node-api-jsi-83cfef428a97627c9185c73da097e42742de56eb\node-api\ -ID:\a\_work\1\s\vnext\Microsoft.ReactNative.Cxx\ -ID:\a\_work\1\s\node_modules\.folly\folly-2023.11.06.00\ -ID:\a\_work\1\s\node_modules\.fmt\fmt-10.1.0\include -ID:\a\_work\1\s\node_modules\react-native\ReactCommon -ID:\a\_work\1\s\node_modules\react-native\ReactCommon\callinvoker -ID:\a\_work\1\s\node_modules\react-native\ReactCommon\jsi -ID:\a\_work\1\s\node_modules\react-native\ReactCommon\jsiexecutor -ID:\a\_work\1\s\node_modules\react-native\ReactCommon\react\nativemodule\core -ID:\a\_work\1\s\node_modules\react-native\ReactCommon\react\nativemodule\samples -ID:\a\_work\1\s\node_modules\react-native\ReactCommon\react\bridging -ID:\a\_work\1\s\node_modules\react-native\ReactCommon\reactperflogger -ID:\a\_work\1\s\node_modules\react-native\ReactCommon\runtimeexecutor -ID:\a\_work\_temp\.nuget\packages\microsoft.windowsappsdk\1.6.240923002\include\ -ID:\a\_work\_temp\.nuget\packages\microsoft.windowsappsdk\1.6.240923002\include -ID:\a\_work\_temp\.nuget\packages\microsoft.windowsappsdk\1.6.240923002\include -ID:\a\_work\1\s\vnext\Shared\ -ID:\a\_work\1\s\node_modules\.node-api-jsi\node-api-jsi-83cfef428a97627c9185c73da097e42742de56eb\src -ID:\a\_work\1\s\vnext\Chakra\ -ID:\a\_work\1\s\vnext\Mso\ -I"D:\a\_work\1\s\vnext\build\x64\Debug\React.Windows.Desktop\Generated Files\\" -I"D:\a\_work\1\s\vnext\build\x64\Debug\React.Windows.Desktop\Generated Files\\" -ID:\a\_work\_temp\.nuget\packages\microsoft.javascript.hermes\0.1.23\build\native\include -Zi -JMC -nologo -W4 -WX -diagnostics:column -Od -D__unused=[[maybe_unused]] -DFOLLY_MOBILE=1 -DBOOST_ASIO_HAS_IOCP -D_WINSOCK_DEPRECATED_NO_WARNINGS -D_WIN32_WINNT=_WIN32_WINNT_WIN10 -DWIN32 -DCORE_ABI -D_WINDOWS -DFOLLY_CFG_NO_COROUTINES -DFOLLY_NO_CONFIG -DNOMINMAX -D_HAS_AUTO_PTR_ETC -DRN_PLATFORM=win32 -DRN_EXPORT= -DJSI_EXPORT= -D_SILENCE_CXX20_U8PATH_DEPRECATION_WARNING -DUSE_WINUI3 -D_UNICODE -DUNICODE -DRNW_VERSION=0.77.0-preview.2 -DRNW_MAJOR=0 -DRNW_MINOR=77 -DRNW_PATCH=0 -DRNW_COMMITID=e70d4987c0929e4eb4125342c53cf6566e19af00 -DWINRT_LEAN_AND_MEAN -D_SILENCE_ALL_CXX17_DEPRECATION_WARNINGS -DBOOST_ALL_NO_LIB=1 -DBOOST_ERROR_CODE_HEADER_ONLY -DBOOST_NO_RTTI -DBOOST_NO_TYPEID -DBOOST_SYSTEM_SOURCE -DGTEST_HAS_RTTI=0 -DUSE_EDGEMODE_JSRT -DWIN32_LEAN_AND_MEAN -DCPPWINRT_USE_LOADLIBRARYEXW -D_DEBUG -DDEBUG -D_DISABLE_CONSTEXPR_MUTEX_CONSTRUCTOR -DJSI_VERSION=11 -DUSE_FABRIC -DUSE_V8 -DUSE_HERMES -DENABLE_JS_SYSTRACE_TO_ETW -DWITH_FBSYSTRACE -DENABLE_ETW_TRACING -DRNW_NEW_ARCH -Gm- -EHs -EHc -RTC1 -MDd -GS -guard:cf -fp:precise -Qspectre -Zc:wchar_t -Zc:forScope -Zc:inline -GR -std:c++20 -Yupch.h -FpD:\a\_work\1\s\vnext\build\x64\Debug\React.Windows.Desktop\React.Windows.Desktop.pch -external:W4 -Gd -TP -wd4244 -wd4715 -wd4456 -wd4702 -wd4458 -wd4068 -wd4100 -wd4101 -wd4127 -wd4189 -wd4290 -wd4309 -wd4324 -wd5205 -wd4251 -FIpch.h -FC -errorreport:queue -utf-8 -Zc:strictStrings -await -bigobj -ID:\a\_work\1\s\vnext\ -ID:\a\_work\1\s\vnext\Common -ID:\a\_work\1\s\vnext\stubs -ID:\a\_work\1\s\vnext\Desktop -ID:\a\_work\1\s\vnext\Shared -ID:\a\_work\1\s\vnext\Shared\tracing -ID:\a\_work\1\s\vnext\include\Shared -I"c:\program files\microsoft visual studio\2022\enterprise\VC\Tools\MSVC\14.42.34433\include" -I"c:\program files\microsoft visual studio\2022\enterprise\VC\Tools\MSVC\14.42.34433\atlmfc\include" -I"c:\program files\microsoft visual studio\2022\enterprise\VC\Auxiliary\VS\include" -I"C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt" -I"c:\program files\microsoft visual studio\2022\enterprise\VC\Auxiliary\VS\UnitTest\include" -I"C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um" -I"C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared" -I"C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\winrt" -ID:\a\_work\1\s\vnext\Desktop\PreventSdkUapPropsAssignment -I"C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8.1\Include\um" -external:I"c:\program files\microsoft visual studio\2022\enterprise\VC\Tools\MSVC\14.42.34433\include" -external:I"c:\program files\microsoft visual studio\2022\enterprise\VC\Tools\MSVC\14.42.34433\atlmfc\include" -external:I"c:\program files\microsoft visual studio\2022\enterprise\VC\Auxiliary\VS\include" -external:I"C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt" -external:I"c:\program files\microsoft visual studio\2022\enterprise\VC\Auxiliary\VS\UnitTest\include" -external:I"C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um" -external:I"C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared" -external:I"C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\winrt" -external:ID:\a\_work\1\s\vnext\Desktop\PreventSdkUapPropsAssignment -external:I"C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8.1\Include\um" -X
Modules triggering this check were:
NativeDOM.obj (React.Windows.Desktop.lib) [Explicitly disabled warnings: 4244]

##[warning]2. BinSkim Warning BA2024 - File: vnext/target/x64/Debug/React.Windows.Desktop.DLL/react-native-win32.dll.  
Signature: 034ef085bf4c54cdc154919814e75380493511b0c06d6adec5689343088aeed6
Tool: BinSkim: Rule: BA2024 (EnableSpectreMitigations). https://github.com/microsoft/binskim/blob/main/docs/BinSkimRules.md#rule-BA2024EnableSpectreMitigations
'react-native-win32.dll' was compiled with one or more modules that do not enable code generation mitigations for speculative execution side-channel attack (Spectre) vulnerabilities. Spectre attacks can compromise hardware-based isolation, allowing non-privileged users to retrieve potentially sensitive data from the CPU cache. To resolve the issue, provide the /Qspectre switch on the compiler command-line (or /d2guardspecload in cases where your compiler supports this switch and it is not possible to update to a toolset that supports /Qspectre). This warning should be addressed for code that operates on data that crosses a trust boundary and that can affect execution, such as parsing untrusted file inputs or processing query strings of a web request.
The following modules were compiled with a toolset that supports /Qspectre but the switch was not enabled on the command-line:
MSVCRTD.lib,cxx,19.42.34321.1 (argv_mode.obj,default_local_stdio_options.obj,delete_array.obj,delete_array_size.obj,delete_scalar.obj,delete_scalar_size.obj,dll_dllmain.obj,dll_dllmain_stub.obj,ehvccctr.obj,ehvecctr.obj,ehvecdtr.obj,error.obj,fltused.obj,gshandler.obj,gshandlereh4.obj,gshandlerseh.obj,init.obj,initializers.obj,initsect.obj,new_array.obj,new_scalar.obj,new_scalar_nothrow.obj,pdblkup.obj,stack.obj,std_nothrow.obj,std_type_info_static.obj,thread_safe_statics.obj,throw_bad_alloc.obj,tlsdyn.obj,tlssup.obj,tncleanup.obj,ucrt_stubs.obj,userapi.obj,utility.obj,utility_desktop.obj)
MSVCRTD.lib,c,19.42.34321.1 (cpu_disp.obj,debugger_jmc.obj,dyn_tls_init.obj,gs_cookie.obj,gs_report.obj,gs_support.obj,guard_support.obj,loadcfg.obj,ucrt_detection.obj)
msvcprtd.lib,cxx,19.42.34321.1 (charconv.obj,filesystem.obj,locale0_implib.obj,sharedmutex.obj,syserror_import_lib.obj,vector_algorithms.obj,xonce2.obj)
Delayimp.lib,cxx,19.42.34321.1 (delaygv1.obj,delayhk1.obj,delayhk2.obj,delayhlp.obj)
vcruntimed.lib,cxx,19.42.34321.1 (softmemtag.obj)
```

### Steps To Reproduce

See publish logs.

### Expected Results

_No response_

### CLI version

npx @react-native-community/cli -v

### Environment

```markdown
npx @react-native-community/cli info
```

### Community Modules

_No response_

### Target Platform Version

None

### Target Device(s)

_No response_

### Visual Studio Version

None

### Build Configuration

None

### Snack, code example, screenshot, or link to a repository

_No response_

--- a/change/react-native-windows-30444064-215c-45d3-aaa7-9c10ae68e203.json
+++ b/change/react-native-windows-30444064-215c-45d3-aaa7-9c10ae68e203.json
@@ -1,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "Stop disabling warning 4244 because BinSkim fails if it sees one",
+  "packageName": "react-native-windows",
+  "email": "jthysell@microsoft.com",
+  "dependentChangeType": "patch"
+}

--- a/vnext/Microsoft.ReactNative/Fabric/Composition/ContentIslandComponentView.h
+++ b/vnext/Microsoft.ReactNative/Fabric/Composition/ContentIslandComponentView.h
@@ -13,7 +13,7 @@
 #include "CompositionViewComponentView.h"
 
 #pragma warning(push)
-#pragma warning(disable : 4244 4305)
+#pragma warning(disable : 4305)
 #include <react/renderer/components/view/ViewProps.h>
 #pragma warning(pop)
 #include "Composition.ContentIslandComponentView.g.h"

--- a/vnext/Microsoft.ReactNative/Fabric/Composition/ImageComponentView.h
+++ b/vnext/Microsoft.ReactNative/Fabric/Composition/ImageComponentView.h
@@ -17,7 +17,7 @@
 #include "ImageResponseImage.h"
 
 #pragma warning(push)
-#pragma warning(disable : 4244 4305)
+#pragma warning(disable : 4305)
 #include <react/renderer/components/view/ViewProps.h>
 #pragma warning(pop)
 #include "Composition.ImageComponentView.g.h"

--- a/vnext/ReactCommon/TEMP_UntilReactCommonUpdate/react/nativemodule/dom/NativeDOM.cpp
+++ b/vnext/ReactCommon/TEMP_UntilReactCommonUpdate/react/nativemodule/dom/NativeDOM.cpp
@@ -1,0 +1,358 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "NativeDOM.h"
+#include <react/renderer/components/root/RootShadowNode.h>
+#include <react/renderer/dom/DOM.h>
+#include <react/renderer/uimanager/PointerEventsProcessor.h>
+#include <react/renderer/uimanager/UIManagerBinding.h>
+
+#ifdef RN_DISABLE_OSS_PLUGIN_HEADER
+#include "Plugins.h"
+#endif
+
+std::shared_ptr<facebook::react::TurboModule> NativeDOMModuleProvider(
+    std::shared_ptr<facebook::react::CallInvoker> jsInvoker) {
+  return std::make_shared<facebook::react::NativeDOM>(std::move(jsInvoker));
+}
+
+namespace facebook::react {
+
+#pragma mark - Private helpers
+
+static RootShadowNode::Shared getCurrentShadowTreeRevision(
+    facebook::jsi::Runtime& runtime,
+    SurfaceId surfaceId) {
+  auto& uiManager =
+      facebook::react::UIManagerBinding::getBinding(runtime)->getUIManager();
+  auto shadowTreeRevisionProvider = uiManager.getShadowTreeRevisionProvider();
+  return shadowTreeRevisionProvider->getCurrentRevision(surfaceId);
+}
+
+static facebook::react::PointerEventsProcessor&
+getPointerEventsProcessorFromRuntime(facebook::jsi::Runtime& runtime) {
+  return facebook::react::UIManagerBinding::getBinding(runtime)
+      ->getPointerEventsProcessor();
+}
+
+static std::vector<facebook::jsi::Value>
+getArrayOfInstanceHandlesFromShadowNodes(
+    const ShadowNode::ListOfShared& nodes,
+    facebook::jsi::Runtime& runtime) {
+  // JSI doesn't support adding elements to an array after creation,
+  // so we need to accumulate the values in a vector and then create
+  // the array when we know the size.
+  std::vector<facebook::jsi::Value> nonNullInstanceHandles;
+  nonNullInstanceHandles.reserve(nodes.size());
+  for (const auto& shadowNode : nodes) {
+    auto instanceHandle = (*shadowNode).getInstanceHandle(runtime);
+    if (!instanceHandle.isNull()) {
+      nonNullInstanceHandles.push_back(std::move(instanceHandle));
+    }
+  }
+
+  return nonNullInstanceHandles;
+}
+
+#pragma mark - NativeDOM
+
+NativeDOM::NativeDOM(std::shared_ptr<CallInvoker> jsInvoker)
+    : NativeDOMCxxSpec(std::move(jsInvoker)) {}
+
+jsi::Value NativeDOM::getParentNode(
+    jsi::Runtime& rt,
+    jsi::Value shadowNodeValue) {
+  auto shadowNode = shadowNodeFromValue(rt, shadowNodeValue);
+  auto currentRevision =
+      getCurrentShadowTreeRevision(rt, shadowNode->getSurfaceId());
+  if (currentRevision == nullptr) {
+    return jsi::Value::undefined();
+  }
+
+  auto parentShadowNode = dom::getParentNode(currentRevision, *shadowNode);
+  if (parentShadowNode == nullptr) {
+    return jsi::Value::undefined();
+  }
+
+  return parentShadowNode->getInstanceHandle(rt);
+}
+
+std::vector<jsi::Value> NativeDOM::getChildNodes(
+    jsi::Runtime& rt,
+    jsi::Value shadowNodeValue) {
+  auto shadowNode = shadowNodeFromValue(rt, shadowNodeValue);
+  auto currentRevision =
+      getCurrentShadowTreeRevision(rt, shadowNode->getSurfaceId());
+  if (currentRevision == nullptr) {
+    return std::vector<jsi::Value>{};
+  }
+
+  auto childNodes = dom::getChildNodes(currentRevision, *shadowNode);
+  return getArrayOfInstanceHandlesFromShadowNodes(childNodes, rt);
+}
+
+bool NativeDOM::isConnected(jsi::Runtime& rt, jsi::Value shadowNodeValue) {
+  auto shadowNode = shadowNodeFromValue(rt, shadowNodeValue);
+  auto currentRevision =
+      getCurrentShadowTreeRevision(rt, shadowNode->getSurfaceId());
+  if (currentRevision == nullptr) {
+    return false;
+  }
+
+  return dom::isConnected(currentRevision, *shadowNode);
+}
+
+double NativeDOM::compareDocumentPosition(
+    jsi::Runtime& rt,
+    jsi::Value shadowNodeValue,
+    jsi::Value otherShadowNodeValue) {
+  auto shadowNode = shadowNodeFromValue(rt, shadowNodeValue);
+  auto otherShadowNode = shadowNodeFromValue(rt, otherShadowNodeValue);
+  auto currentRevision =
+      getCurrentShadowTreeRevision(rt, shadowNode->getSurfaceId());
+  if (otherShadowNode == nullptr || currentRevision == nullptr) {
+    return 0;
+  }
+
+  return dom::compareDocumentPosition(
+      currentRevision, *shadowNode, *otherShadowNode);
+}
+
+std::string NativeDOM::getTextContent(
+    jsi::Runtime& rt,
+    jsi::Value shadowNodeValue) {
+  auto shadowNode = shadowNodeFromValue(rt, shadowNodeValue);
+  auto currentRevision =
+      getCurrentShadowTreeRevision(rt, shadowNode->getSurfaceId());
+  if (currentRevision == nullptr) {
+    return "";
+  }
+
+  return dom::getTextContent(currentRevision, *shadowNode);
+}
+
+std::tuple<
+    /* x: */ double,
+    /* y: */ double,
+    /* width: */ double,
+    /* height: */ double>
+NativeDOM::getBoundingClientRect(
+    jsi::Runtime& rt,
+    jsi::Value shadowNodeValue,
+    bool includeTransform) {
+  auto shadowNode = shadowNodeFromValue(rt, shadowNodeValue);
+  auto currentRevision =
+      getCurrentShadowTreeRevision(rt, shadowNode->getSurfaceId());
+  if (currentRevision == nullptr) {
+    return {0, 0, 0, 0};
+  }
+
+  auto domRect = dom::getBoundingClientRect(
+      currentRevision, *shadowNode, includeTransform);
+
+  return std::tuple{domRect.x, domRect.y, domRect.width, domRect.height};
+}
+
+std::tuple<
+    /* offsetParent: */ jsi::Value,
+    /* top: */ double,
+    /* left: */ double>
+NativeDOM::getOffset(jsi::Runtime& rt, jsi::Value shadowNodeValue) {
+  auto shadowNode = shadowNodeFromValue(rt, shadowNodeValue);
+  auto currentRevision =
+      getCurrentShadowTreeRevision(rt, shadowNode->getSurfaceId());
+  if (currentRevision == nullptr) {
+    return {jsi::Value::undefined(), 0, 0};
+  }
+
+  auto domOffset = dom::getOffset(currentRevision, *shadowNode);
+
+  return std::tuple{
+      domOffset.offsetParent == nullptr
+          ? jsi::Value::undefined()
+          : domOffset.offsetParent->getInstanceHandle(rt),
+      domOffset.top,
+      domOffset.left};
+}
+
+std::tuple</* scrollLeft: */ double, /* scrollTop: */ double>
+NativeDOM::getScrollPosition(jsi::Runtime& rt, jsi::Value shadowNodeValue) {
+  auto shadowNode = shadowNodeFromValue(rt, shadowNodeValue);
+  auto currentRevision =
+      getCurrentShadowTreeRevision(rt, shadowNode->getSurfaceId());
+  if (currentRevision == nullptr) {
+    return {0, 0};
+  }
+
+  auto domPoint = dom::getScrollPosition(currentRevision, *shadowNode);
+  return std::tuple{domPoint.x, domPoint.y};
+}
+
+std::tuple</* scrollWidth: */ int, /* scrollHeight */ int>
+NativeDOM::getScrollSize(jsi::Runtime& rt, jsi::Value shadowNodeValue) {
+  auto shadowNode = shadowNodeFromValue(rt, shadowNodeValue);
+  auto currentRevision =
+      getCurrentShadowTreeRevision(rt, shadowNode->getSurfaceId());
+  if (currentRevision == nullptr) {
+    return {0, 0};
+  }
+
+  auto scrollSize = dom::getScrollSize(currentRevision, *shadowNode);
+  return std::tuple{scrollSize.width, scrollSize.height};
+}
+
+std::tuple</* width: */ int, /* height: */ int> NativeDOM::getInnerSize(
+    jsi::Runtime& rt,
+    jsi::Value shadowNodeValue) {
+  auto shadowNode = shadowNodeFromValue(rt, shadowNodeValue);
+  auto currentRevision =
+      getCurrentShadowTreeRevision(rt, shadowNode->getSurfaceId());
+  if (currentRevision == nullptr) {
+    return {0, 0};
+  }
+
+  auto innerSize = dom::getInnerSize(currentRevision, *shadowNode);
+  return std::tuple{innerSize.width, innerSize.height};
+}
+
+std::tuple<
+    /* topWidth: */ int,
+    /* rightWidth: */ int,
+    /* bottomWidth: */ int,
+    /* leftWidth: */ int>
+NativeDOM::getBorderWidth(jsi::Runtime& rt, jsi::Value shadowNodeValue) {
+  auto shadowNode = shadowNodeFromValue(rt, shadowNodeValue);
+  auto currentRevision =
+      getCurrentShadowTreeRevision(rt, shadowNode->getSurfaceId());
+  if (currentRevision == nullptr) {
+    return {0, 0, 0, 0};
+  }
+
+  auto borderWidth = dom::getBorderWidth(currentRevision, *shadowNode);
+  return std::tuple{
+      borderWidth.top, borderWidth.right, borderWidth.bottom, borderWidth.left};
+}
+
+std::string NativeDOM::getTagName(
+    jsi::Runtime& rt,
+    jsi::Value shadowNodeValue) {
+  auto shadowNode = shadowNodeFromValue(rt, shadowNodeValue);
+  return dom::getTagName(*shadowNode);
+}
+
+#pragma mark - Pointer events
+
+bool NativeDOM::hasPointerCapture(
+    jsi::Runtime& rt,
+    jsi::Value shadowNodeValue,
+    double pointerId) {
+  bool isCapturing = getPointerEventsProcessorFromRuntime(rt).hasPointerCapture(
+      // [Windows] Cast to fix https://github.com/microsoft/react-native-windows/issues/14251
+      static_cast<PointerIdentifier>(pointerId), shadowNodeFromValue(rt, shadowNodeValue).get());
+  return isCapturing;
+}
+
+void NativeDOM::setPointerCapture(
+    jsi::Runtime& rt,
+    jsi::Value shadowNodeValue,
+    double pointerId) {
+  getPointerEventsProcessorFromRuntime(rt).setPointerCapture(
+      // [Windows] Cast to fix https://github.com/microsoft/react-native-windows/issues/14251
+      static_cast<PointerIdentifier>(pointerId), shadowNodeFromValue(rt, shadowNodeValue));
+}
+
+void NativeDOM::releasePointerCapture(
+    jsi::Runtime& rt,
+    jsi::Value shadowNodeValue,
+    double pointerId) {
+  getPointerEventsProcessorFromRuntime(rt).releasePointerCapture(
+      // [Windows] Cast to fix https://github.com/microsoft/react-native-windows/issues/14251
+      static_cast<PointerIdentifier>(pointerId), shadowNodeFromValue(rt, shadowNodeValue).get());
+}
+
+#pragma mark - Legacy RN layout APIs
+
+void NativeDOM::measure(
+    jsi::Runtime& rt,
+    jsi::Value shadowNodeValue,
+    jsi::Function callback) {
+  auto shadowNode = shadowNodeFromValue(rt, shadowNodeValue);
+  auto currentRevision =
+      getCurrentShadowTreeRevision(rt, shadowNode->getSurfaceId());
+  if (currentRevision == nullptr) {
+    callback.call(rt, {0, 0, 0, 0, 0, 0});
+    return;
+  }
+
+  auto measureRect = dom::measure(currentRevision, *shadowNode);
+
+  callback.call(
+      rt,
+      {jsi::Value{rt, measureRect.x},
+       jsi::Value{rt, measureRect.y},
+       jsi::Value{rt, measureRect.width},
+       jsi::Value{rt, measureRect.height},
+       jsi::Value{rt, measureRect.pageX},
+       jsi::Value{rt, measureRect.pageY}});
+}
+
+void NativeDOM::measureInWindow(
+    jsi::Runtime& rt,
+    jsi::Value shadowNodeValue,
+    jsi::Function callback) {
+  auto shadowNode = shadowNodeFromValue(rt, shadowNodeValue);
+  auto currentRevision =
+      getCurrentShadowTreeRevision(rt, shadowNode->getSurfaceId());
+  if (currentRevision == nullptr) {
+    callback.call(rt, {0, 0, 0, 0});
+    return;
+  }
+
+  auto rect = dom::measureInWindow(currentRevision, *shadowNode);
+  callback.call(
+      rt,
+      {jsi::Value{rt, rect.x},
+       jsi::Value{rt, rect.y},
+       jsi::Value{rt, rect.width},
+       jsi::Value{rt, rect.height}});
+}
+
+void NativeDOM::measureLayout(
+    jsi::Runtime& rt,
+    jsi::Value shadowNodeValue,
+    jsi::Value relativeToShadowNodeValue,
+    jsi::Function onFail,
+    jsi::Function onSuccess) {
+  auto shadowNode = shadowNodeFromValue(rt, shadowNodeValue);
+  auto relativeToShadowNode =
+      shadowNodeFromValue(rt, relativeToShadowNodeValue);
+  auto currentRevision =
+      getCurrentShadowTreeRevision(rt, shadowNode->getSurfaceId());
+  if (currentRevision == nullptr) {
+    onFail.call(rt);
+    return;
+  }
+
+  auto maybeRect =
+      dom::measureLayout(currentRevision, *shadowNode, *relativeToShadowNode);
+
+  if (!maybeRect) {
+    onFail.call(rt);
+    return;
+  }
+
+  auto rect = maybeRect.value();
+
+  onSuccess.call(
+      rt,
+      {jsi::Value{rt, rect.x},
+       jsi::Value{rt, rect.y},
+       jsi::Value{rt, rect.width},
+       jsi::Value{rt, rect.height}});
+}
+
+} // namespace facebook::react

--- a/vnext/Shared/Shared.vcxitems
+++ b/vnext/Shared/Shared.vcxitems
@@ -555,7 +555,7 @@
     <!-- Uncomment this when we move to a newer JSI -->
     <!-- <CLCompile Include="$(ReactNativeDir)\ReactCommon\react\nativemodule\microtasks\NativeMicrotasks.cpp" /> -->
     <CLCompile Include="$(ReactNativeDir)\ReactCommon\react\nativemodule\featureflags\NativeReactNativeFeatureFlags.cpp" />
-    <CLCompile Include="$(ReactNativeDir)\ReactCommon\react\nativemodule\dom\NativeDOM.cpp" DisableSpecificWarnings="4244;4715;%(DisableSpecificWarnings)" />
+    <CLCompile Include="$(ReactNativeDir)\ReactCommon\react\nativemodule\dom\NativeDOM.cpp" DisableSpecificWarnings="4715;%(DisableSpecificWarnings)" />
     <CLCompile Include="$(ReactNativeDir)\ReactCommon\react\nativemodule\idlecallbacks\NativeIdleCallbacks.cpp" />
     <ClCompile Include="$(ReactNativeDir)\ReactCommon\react\renderer\attributedstring\AttributedString.cpp" />
     <ClCompile Include="$(ReactNativeDir)\ReactCommon\react\renderer\attributedstring\AttributedStringBox.cpp" DisableSpecificWarnings="4715;%(DisableSpecificWarnings)" />

--- a/vnext/overrides.json
+++ b/vnext/overrides.json
@@ -129,6 +129,13 @@
       "issue": 14128
     },
     {
+      "type": "patch",
+      "file": "ReactCommon/TEMP_UntilReactCommonUpdate/react/nativemodule/dom/NativeDOM.cpp",
+      "baseFile": "packages/react-native/ReactCommon/react/nativemodule/dom/NativeDOM.cpp",
+      "baseHash": "d1424b3fbb7b7edd73c7a3e3ceb09b47c3953051",
+      "issue": 14257
+    },
+    {
       "type": "derived",
       "file": "ReactCommon/TEMP_UntilReactCommonUpdate/react/renderer/componentregistry/ComponentDescriptorRegistry.cpp",
       "baseFile": "packages/react-native/ReactCommon/react/renderer/componentregistry/ComponentDescriptorRegistry.cpp",

